Lab1/app/src/main/java/nm/lab1/MainActivity.java

```diff
@@ -0,0 +1,144 @@
+    //long click listener
+        list.setOnItemLongClickListener(new AdapterView.OnItemLongClickListener() {
+            @Override
+            public boolean onItemLongClick(AdapterView<?> parent, View view, int position, long id) {
+                //remove item in listarray at position
+                doAlertDialog(arrayList, position, listAdapter); //brings up alert dialog
+                return true;
+            }
+        });
+    //short click listener
+        list.setOnItemClickListener(new AdapterView.OnItemClickListener() {
+            @Override
+            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
+                //arrayList.set(position, "INPUT");
+                doInputAlertDialog(arrayList, position, listAdapter);
+            }
+        });
+    }
+
+    //alert dialog for editing
+    private void doInputAlertDialog(final ArrayList<String> arrayList, final int position, final ArrayAdapter<String> listAdapter) {
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+        alertDialogBuilder.setTitle("Edit item: ");
+        final EditText input = new EditText(this);
+        alertDialogBuilder.setView(input);
+
+        //alert buttons
+        alertDialogBuilder.setPositiveButton("OK", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                String text = input.getText().toString();
+                arrayList.set(position, text);
+                listAdapter.notifyDataSetChanged();
+            }
+        });
+        alertDialogBuilder.setNegativeButton("Cancel", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                dialog.cancel();
+            }
+        });
+        alertDialogBuilder.show();
+    }
+    
+    //alert dialog for deleting item
+    private void doAlertDialog(final ArrayList<String> arrayList, final int position, final ArrayAdapter<String> listAdapter){
+        //alert dialog
+        AlertDialog.Builder alertDialogBuilder = new AlertDialog.Builder(this);
+        alertDialogBuilder.setMessage("Delete this item?");
+
+        //alert buttons
+        alertDialogBuilder.setPositiveButton("Yes", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog, int which) {
+                Toast.makeText(MainActivity.this, "Item removed", Toast.LENGTH_LONG).show();
+                arrayList.remove(position); //remove choice
+                listAdapter.notifyDataSetChanged();
+            }
+        });
+        alertDialogBuilder.setNegativeButton("No", new DialogInterface.OnClickListener() {
+            @Override
+            public void onClick(DialogInterface dialog2, int which2) {
```

Comment:
Because of scope this doesn't have to be named dialog2 and which2. These names are only recognized within each method so they can be they same and they will not conflict. More on scope here http://programmers.stackexchange.com/questions/141133/what-is-meant-by-scope-of-a-variable